helper_test.go

```diff
@@ -88,6 +89,85 @@ func fetchTestAccountDetails(t *testing.T, client *Client) *TestAccountDetails {
 	return _testAccountDetails
 }
 
+func createAgent(t *testing.T, client *Client, org *Organization, agentPool *AgentPool, agentPoolToken *AgentToken) (*Agent, *AgentPool, func()) {
```

Comment:
Okay so this method signature is a bit long (apologies for the pickiness):

So I understand you've expanded the interface so that you wouldn't have to call:

```go
agentPool, agentPoolCleanup := createAgentPool(stuff, here)
```

within each test. But if I'm reading for the first time what this helper method is doing: 
```go
agent, _, agentCleanup := createAgent(t, client, nil, nil, nil)
```
I wouldn't really be sure what's going on, and I also wouldn't be able to deduce that `createAgent()` also creates an agent pool. 

Here's what I'm proposing:

```go
func createAgent(t *testing.T, client *Client, org *Organization, pool *AgentPool) (*Agent, func()) {}
```

We don't need to pass an `AgentToken` since we only ever use it upon creation. I guess there may be some subsequent test added to validate the token, but I don't think that falls under the scope of the Agents API (we know a token is valid if the agent is registered).

So we can simply create a new token inside of `createAgent()` and include its deletion in the cleanup function. 

The second thing is to remove `AgentPool` from the return value. If I wanted to use the `AgentPool`'s values in tests, I should create the agent pool beforehand and pass it into `createAgent()` otherwise why would we allow passing an agent pool as a parameter if we can simply use the one the method creates?